### PR TITLE
improvement(directives): add type declaration

### PR DESCRIPTION
**Pull request by xiaoxian521**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow Element's contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by xiaoxian521** [comment]
fix: https://github.com/element-plus/element-plus/issues/3313

File: packages/directives/resize/index.ts [hunk 1]
```diff
@@ -3,16 +3,16 @@ import {
...
 const Resize: ObjectDirective = {
-  beforeMount(el, binding) {
+  beforeMount(el: HTMLElement | any, binding: DirectiveBinding) {
     el._handleResize = () => {
       el && binding.value?.(el)
     }
     addResizeListener(el, el._handleResize)
   },
-  beforeUnmount(el) {
+  beforeUnmount(el: HTMLElement | any) {
```
Comment (sxzz) [inline packages/directives/resize/index.ts:15]: just use `any`

File: packages/directives/resize/index.ts [hunk 2]
```diff
@@ -3,16 +3,16 @@ import {
   removeResizeListener,
 } from '@element-plus/utils/resize-event'
 
-import type { ObjectDirective } from 'vue'
+import type { ObjectDirective, DirectiveBinding } from 'vue'
 
 const Resize: ObjectDirective = {
-  beforeMount(el, binding) {
+  beforeMount(el: HTMLElement | any, binding: DirectiveBinding) {
```
Comment (sxzz) [inline packages/directives/resize/index.ts:9]: just use `any`

File: packages/directives/resize/index.ts [hunk 3]
```diff
@@ -3,16 +3,20 @@ import {
...
+import type { ObjectDirective, DirectiveBinding } from 'vue'
+
+declare interface ResizeEl {
+  _handleResize: () => void
+}
 
 const Resize: ObjectDirective = {
-  beforeMount(el, binding) {
+  beforeMount(el: ResizeEl, binding: DirectiveBinding) {
     el._handleResize = () => {
```
Comment (sxzz) [inline packages/directives/resize/index.ts:14]: `el` is `HTMLElement`, how about
```suggestion
  beforeMount(el: HTMLElement, binding: DirectiveBinding) {
   (el as any)._handleResize = () => {
```

Comment (xiaoxian521) [reply packages/directives/resize/index.ts:14]: HTMLElement does not exist _handleResize function

Comment (xiaoxian521) [reply packages/directives/resize/index.ts:14]:
```
(el as any) 
```
In this case, it's still any type.

File: packages/directives/resize/index.ts [hunk 4]
```diff
@@ -3,16 +3,20 @@ import {
   removeResizeListener,
 } from '@element-plus/utils/resize-event'
 
-import type { ObjectDirective } from 'vue'
+import type { ObjectDirective, DirectiveBinding } from 'vue'
+
+declare interface ResizeEl {
+  _handleResize: () => void
+}
```
Comment (sxzz) [inline packages/directives/resize/index.ts:10]: Emmm, or like this?
```suggestion
declare interface ResizeEl extends HTMLElement {
  _handleResize?: () => void
}
```

Comment (xiaoxian521) [reply packages/directives/resize/index.ts:10]: haha Perfect

**Review by sxzz — APPROVED** [review]

**Comment by sxzz** [comment]
LGTM. Thank you.

**Comment by xiaoxian521** [comment]
> LGTM. Thank you.

Live more and more powerful haha

**Pull request merged**